identity-service/test/configTest.js

```diff
@@ -0,0 +1,150 @@
+const assert = require('assert')
+const config = require('../src/config')
+const { generateRandomUUID, generateRandomPort, generateRandomNaturalNumber, generateRandomBoolean, generateRandomLogLevel } = require('./lib/generateRandomValues')
+const convict = require('convict')
+
+describe('test txRelay functions', function () {
+  it('env key and object key names are equal', function () {
+    const schema = config.getSchema().properties
+
+    for (var key in schema) {
+      assert(key === schema[key]['env'], `The key ${key} and the env key ${schema[key]['env']} do not match.`)
+    }
+  })
+
+  it('[pass] Adding a number with format as type string will cause config to take its string value', function () {
+    const c = convict({
+      randomKey: {
+        format: String,
+        default: null,
+        env: 'randomKey'
+      }
+    })
+
+    process.env['randomKey'] = 123
+    assert(process.env['randomKey'] === '123')
+
+    c.load({})
+
+    assert(c.get('randomKey') === '123')
+
+    try {
+      c.validate()
+    } catch (e) {
+      assert.fail(e)
+    }
+  })
+
+  it('[pass] Adding a number with format as type nat will cause config to take its numeric value', function () {
+    const c = convict({
+      randomKey: {
+        format: 'nat',
+        default: null,
+        env: 'randomKey'
+      }
+    })
+
+    process.env['randomKey'] = '123'
+
+    c.load({})
+
+    assert(c.get('randomKey') === 123)
+
+    try {
+      c.validate()
+    } catch (e) {
+      assert.fail(e)
+    }
+  })
+
+  it('setting a value through env key, and ensuring the new value is set in config', function () {
+    const schema = config.getSchema().properties
+
+    for (var key in schema) {
+      let oldValue = config.get(key)
+      let newValue
+
+      let format = schema[key].format.toString()
+
+      // setting invalid values
+      newValue = getBadValue(format, newValue)
+
+      if (newValue) {
+        setAndValidateEnvVar(key, newValue)
+
+        // Reload config to grab env vars
+        config.load({})
+
+        assert.throws(() => {
+          config.validate()
+          assert.fail('Did not properly validate config')
+        }, Error)
+      }
+
+      newValue = getValidValue(format, newValue)
```

Comment:
this is a NIT but let us name the valid and invalid values separately...this is pretty confusing imo